src/npyffi/objects.rs

```diff
@@ -44,6 +57,139 @@ pub struct PyArray_Descr {
     pub hash: npy_hash_t,
 }
 
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct _PyArray_DescrNumPy2 {
+    pub ob_base: PyObject,
+    pub typeobj: *mut PyTypeObject,
+    pub kind: c_char,
+    pub type_: c_char,
+    pub byteorder: c_char,
+    pub _former_flags: c_char,
+    pub type_num: c_int,
+    pub flags: npy_uint64,
+    pub elsize: npy_intp,
+    pub alignment: npy_intp,
+    pub metadata: *mut PyObject,
+    pub hash: npy_hash_t,
+    pub reserved_null: [*mut std::ffi::c_void; 2],
+}
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+struct _PyArray_LegacyDescr {
+    pub ob_base: PyObject,
+    pub typeobj: *mut PyTypeObject,
+    pub kind: c_char,
+    pub type_: c_char,
+    pub byteorder: c_char,
+    pub _former_flags: c_char,
+    pub type_num: c_int,
+    pub flags: npy_uint64,
+    pub elsize: npy_intp,
+    pub alignment: npy_intp,
+    pub metadata: *mut PyObject,
+    pub hash: npy_hash_t,
+    pub reserved_null: [*mut std::ffi::c_void; 2],
+    pub subarray: *mut PyArray_ArrayDescr,
+    pub fields: *mut PyObject,
+    pub names: *mut PyObject,
+    pub c_metadata: *mut NpyAuxData,
+}
+
+#[allow(non_snake_case)]
+#[inline(always)]
+pub unsafe fn PyDataType_ISLEGACY(dtype: *const PyArray_Descr) -> bool {
+    (*dtype).type_num < NPY_TYPES::NPY_VSTRING as i32 && (*dtype).type_num >= 0
+}
+
+#[allow(non_snake_case)]
+#[inline(always)]
+pub unsafe fn PyDataType_SET_ELSIZE<'py>(
+    py: Python<'py>,
+    dtype: *mut PyArray_Descr,
+    size: npy_intp,
+) {
+    let api_version = *API_VERSION.get(py).expect("API_VERSION is initialized");
+    if api_version < API_VERSION_2_0 {
+        unsafe {
+            (*(dtype as *mut PyArray_DescrProto)).elsize = size as c_int;
+        }
+    } else {
+        unsafe {
+            (*(dtype as *mut _PyArray_DescrNumPy2)).elsize = size;
+        }
+    }
+}
+
+#[allow(non_snake_case)]
+#[inline(always)]
+pub unsafe fn PyDataType_FLAGS<'py>(py: Python<'py>, dtype: *const PyArray_Descr) -> npy_uint64 {
+    let api_version = *API_VERSION.get(py).expect("API_VERSION is initialized");
+    if api_version < API_VERSION_2_0 {
+        unsafe { (*(dtype as *mut PyArray_DescrProto)).flags as c_uchar as npy_uint64 }
+    } else {
+        unsafe { (*(dtype as *mut _PyArray_DescrNumPy2)).flags }
+    }
+}
+
+macro_rules! define_descr_accessor {
+    ($name:ident, $property:ident, $type:ty, $legacy_only:literal, $zero:expr) => {
```

Comment:
Please rename `$zero` to `$default` or `$fallback` or something like that.

Reply:
Done in #442.